src/Pkb/Parsobober.Pkb.Relations/Implementations/UsesRelation.cs

```diff
@@ -62,4 +64,40 @@ public void SetUses(TreeNode user, TreeNode variable)
 
         _usesDictionary.Add(user.LineNumber, [variable.Attribute]);
     }
+
+    public IEnumerable<Variable> GetVariables<T>() where T : IUsesAccessor.IRequest
+    {
+
+        return _usesDictionary
+            .Where(statement => programContext.StatementsDictionary[statement.Key].IsType<T>())
+            .SelectMany(statement => statement.Value)
+            .Distinct()
+            .Select(variable => programContext.VariablesDictionary[variable].ToVariable());
+    }
+
+    public IEnumerable<Variable> GetVariables(int lineNumber)
+    {
+        if (_usesDictionary.TryGetValue(lineNumber, out var variableList))
+        {
+            return variableList.Select(variableName => new Variable(variableName));
+        }
+        else
+        {
+            return [];
```

Comment:
Możnaby zwrócić puste Enumerable
```suggestion
            return Enumerable.Empty<Variable>();
```